lib/event_tracer/datadog_logger.rb

```diff
@@ -0,0 +1,81 @@
+require_relative '../event_tracer'
+require_relative './basic_decorator'
+# NOTES
+# Datadog interface to send our usual actions
+# BasicDecorator adds a transparent interface on top of the datadog interface
+#
+# Usage: EventTracer.register :datadog, EventTracer::DataDogLogger.new(DataDog)
+#        data_dog_logger.info datadog: { increment: { counter_1: 1, counter_2: 2 }, set: { gauge_1: 1 } }
+#        data_dog_logger.info datadog: { increment: { counter_1: { value: 1, tags: ['tag1, tag2']} } }
+
+module EventTracer
+  class DatadogLogger < BasicDecorator
+
+    class InvalidTagError < StandardError; end
+
+    SUPPORTED_METRICS ||= %i[increment set distribution gauge histogram].freeze
+
+    LOG_TYPES.each do |log_type|
+      define_method log_type do |**args|
+        return LogResult.new(false, 'Invalid datadog config') unless args[:datadog]&.is_a?(Hash)
+
+        applied_metrics(args[:datadog]).each do |metric|
+          metric_args = args[:datadog][metric]
+          return LogResult.new(false, "Datadog metric #{metric} invalid") unless metric_args.is_a?(Hash)
+
+          send_metric metric, metric_args
+        rescue InvalidTagError => e
```

Comment:
If the tag is invalid, should we just raise them instead of failing silently? It would help us notice our mistake earlier.

Reply:
noted, will update